.github/workflows/build.yml

```diff
@@ -0,0 +1,266 @@
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+jobs:
+  linux:
+    name: linux
+    runs-on: ubuntu-latest
+    container: lmmsci/linux.gcc:18.04
+    env:
+      CMAKE_OPTS: >-
+        -DUSE_WERROR=ON
+        -DCMAKE_BUILD_TYPE=RelWithDebInfo
+        -DUSE_COMPILE_CACHE=ON
+      CCACHE_MAXSIZE: 500M
+      MAKEFLAGS: -j2
+    steps:
+      - name: Update and configure Git
+        run: |
+          add-apt-repository ppa:git-core/ppa
+          apt-get update
+          apt-get --yes install git
+          git config --global --add safe.directory /__w/lmms/lmms
+      - name: Check out
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          submodules: recursive
+      - name: Cache ccache data
+        uses: actions/cache@v3
+        with:
+          key: ccache-${{ github.job }}-${{ github.ref }}-${{ github.run_id }}
+          restore-keys: |
+            ccache-${{ github.job }}-${{ github.ref }}-
+            ccache-${{ github.job }}-
+          path: ~/.ccache
+      - name: Configure
+        run: |
+          source /opt/qt5*/bin/qt5*-env.sh || true
+          mkdir build && cd build
+          cmake .. $CMAKE_OPTS -DCMAKE_INSTALL_PREFIX=./install
+      - name: Build
+        run: cmake --build build
+      - name: Build tests
+        run: cmake --build build --target tests
+      - name: Run tests
+        run: build/tests/tests
+      - name: Package
+        run: |
+          cmake --build build --target install
+          cmake --build build --target appimage \
+            || (cat build/appimage.log && false)
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: linux
+          path: build/lmms-*.AppImage
+      - name: Print ccache statistics
+        run: |
+          echo "[ccache config]"
+          ccache -p
+          echo "[ccache stats]"
+          ccache -s
+  macos:
+    name: macos
+    runs-on: macos-11
+    env:
+      CMAKE_OPTS: >-
+        -DUSE_WERROR=ON
+        -DCMAKE_BUILD_TYPE=RelWithDebInfo
+        -DUSE_COMPILE_CACHE=ON
+      CCACHE_MAXSIZE: 500M
+      MAKEFLAGS: -j3
+      DEVELOPER_DIR: /Applications/Xcode_11.7.app/Contents/Developer
+    steps:
+      - name: Check out
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          submodules: recursive
+      - name: Cache ccache data
+        uses: actions/cache@v3
+        with:
+          key: ccache-${{ github.job }}-${{ github.ref }}-${{ github.run_id }}
+          restore-keys: |
+            ccache-${{ github.job }}-${{ github.ref }}-
+            ccache-${{ github.job }}-
+          path: ~/Library/Caches/ccache
+      - name: Install dependencies
+        run: |
+          brew install ccache fftw pkg-config libogg libvorbis lame libsndfile \
+                       libsamplerate jack sdl libgig libsoundio lilv lv2 stk \
+                       fluid-synth portaudio fltk qt5 carla
```

Comment:
Not important, but `qt5` is an alias of `qt@5` now.

Reply:
Do you think it's worth changing? I just copy-pasted this from the CircleCI configuration.

Reply:
I prefer `qt@5` in new projects because it's the standard way to mark formula version in Homebrew, but `qt5` is also acceptable as long as it works.